File: 프로그래머스/3/43162. 네트워크/네트워크.swift
import Foundation

var dict = [Int: [Int]]()
var visited = [Bool]()
func bfs(_ start: Int) {
    var queue: [Int] = [start]
    while !queue.isEmpty {
        let cur = queue.removeFirst()
        visited[cur] = true
        for nxt in dict[cur]! {
            if !visited[nxt] {
                visited[nxt] = true
                queue.append(nxt)
            }
        }
    }
}

func solution(_ n:Int, _ computers:[[Int]]) -> Int {
    var network = 0
    dict = Dictionary(uniqueKeysWithValues: (0..<n).map { ($0, [Int]())})
    visited = Array(repeating: false, count: n)
    
    for i in 0..<n {
        for j in 0..<n {
            if i == j { continue }
            if computers[i][j] == 1 {
                dict[i]?.append(j)    
            }
        }
    }
    for i in 0..<n {
        if !visited[i] {
            bfs(i)
            network += 1
        }
    }

    return network
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque
visited = []
dict = {}

def bfs(start):
    global visited, dict
    queue = deque([start])
    visited[start] = True
    while queue:
        cur = queue.popleft()
        for nxt in dict[cur]:
            if not visited[nxt]:
                visited[nxt] = True
                queue.append(nxt)
                
def solution(n, computers):
    global visited, dict
    network = 0
    visited = [False]*n
    dict = {i: [] for i in range(n)}
    for i in range(len(computers)):
        for j in range(len(computers[i])):
            if i == j:
                continue
            if computers[i][j] == 1:
                dict[i].append(j)

    for i in range(n):
        if not visited[i]:
            bfs(i)
            network += 1
                
    return network